src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageHeader.java

```diff
@@ -37,14 +41,18 @@ public void validate() {
             throw new EdifactValidationException(KEY + ": Attribute sequenceNumber must be between 1 and "
                 + MAX_MESSAGE_SEQUENCE);
         }
+        if (StringUtils.isBlank(messageType)) {
+            throw new EdifactValidationException(KEY + ": Attribute messageType is required");
+        }
     }
 
     public static MessageHeader fromString(final String edifactString) {
         if (!edifactString.startsWith(KEY)) {
             throw new IllegalArgumentException("Can't create " + MessageHeader.class.getSimpleName()
                 + " from " + edifactString);
         }
-        final String[] split = Split.byPlus(edifactString);
-        return new MessageHeader(Long.valueOf(split[1]));
+        final String[] splitByPlus = Split.byPlus(edifactString);
+        final String[] splitByColon = Split.byColon(splitByPlus[2]);
+        return new MessageHeader(Long.valueOf(splitByPlus[1]), splitByColon[INDEX_MESSAGE_TYPE]);
```

Comment:
Should we add another constant field for the index of the sequence number, to make it obvious what we're trying to extract here?
```suggestion
        return new MessageHeader(Long.valueOf(splitByPlus[INDEX_SEQUENCE_NUMBER]), splitByColon[INDEX_MESSAGE_TYPE]);
```

Reply:
Yep!